<launch>

    <arg name="model" default="$(find me134)/urdf/robot.urdf"/>
    <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="me134" type="motor_controller.py" />

    <node name="imu_node"         pkg="me134" type="imu_node.py"      output="screen" />
    <node name="kalman_filter"    pkg="me134" type="kalman_filter.py" output="screen" >
        <param name="segway_tilt_input_topics" value="imu_tilt,lidar_tilt" />
    </node>
    <node name="motor_controller" pkg="me134" type="motor_controller" output="screen" />

    <node name="tilt_pid"  pkg="me134" type="main_controller.py"           output="screen" >
        <param name="Kp" value="5.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />

        <param name="upper_limit" value="10.0" />
        <param name="lower_limit" value="-10.0" />

        <param name="cutoff_frequency" value="20.0" />
        <param name="target_loop_freqency" value="50.0" />
        <param name="min_loop_frequency" value="40.0" />
    </node>

    <node name="sonar_node"       pkg="me134" type="sonar_node.py"    output="screen" />
</launch>
